vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp

```diff
@@ -612,10 +612,18 @@ void TextInputShadowNode::SetText(const winrt::Microsoft::ReactNative::JSValue &
   if (m_mostRecentEventCount == m_nativeEventCount) {
     if (textBox) {
       if (text.Type() == winrt::Microsoft::ReactNative::JSValueType::String) {
+        auto oldCursor = textBox.SelectionStart();
+        auto oldSelectionLength = textBox.SelectionLength();
         auto oldValue = textBox.Text();
         auto newValue = react::uwp::asHstring(text);
         if (oldValue != newValue) {
           textBox.Text(newValue);
+          if (static_cast<uint32_t>(oldCursor) <= newValue.size()) {
+            textBox.SelectionStart(oldCursor);
+          } else {
+            textBox.SelectionStart(newValue.size());
+          }
+          textBox.SelectionLength(oldSelectionLength);
```

Comment:
I don't think we should be preserving SelectionLength. From what I am seeing in ios/android, the cursor position (SelectionStart) stays the same, but any kind of selection is lost on text change.

Reply:
Have you tested it on those platforms? Looking at code might not say everything about native behavior, as is the case with calling SetText() on Windows.

So I wasn't sure about this one., and don't have a strong view. Let's get more opinions and then I can change it perhaps.

Reply:
Removed selection persisting